5calls/app/src/main/java/org/a5calls/android/a5calls/MainActivity.java

```diff
@@ -248,68 +244,105 @@ public void updateIssue(Issue issue) {
         }
 
         @Override
-        public IssueViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
-            RelativeLayout v = (RelativeLayout) LayoutInflater.from(parent.getContext())
-                    .inflate(R.layout.issue_view, parent, false);
-            IssueViewHolder vh = new IssueViewHolder(v);
-            return vh;
-        }
-
-        @Override
-        public void onBindViewHolder(final IssueViewHolder holder, int position) {
-            final Issue issue = mIssues.get(position);
-            holder.name.setText(issue.name);
-            holder.itemView.setOnClickListener(new View.OnClickListener() {
-                @Override
-                public void onClick(View v) {
-                    Intent issueIntent = new Intent(holder.itemView.getContext(),
-                            IssueActivity.class);
-                    issueIntent.putExtra(IssueActivity.KEY_ISSUE, issue);
-                    issueIntent.putExtra(IssueActivity.KEY_ZIP, mZip);
-                    startActivityForResult(issueIntent, ISSUE_DETAIL_REQUEST);
+        public void onBindViewHolder(RecyclerView.ViewHolder viewHolder, int position) {
+            if (position == 0) {
+                final HeaderViewHolder holder = (HeaderViewHolder) viewHolder;
+
+                // Load the zip code the user last used, if any.
+                String code = pref.getString(KEY_USER_ZIP, "");
```

Comment:
I'd prefer not to pass preferences into the Adapter, it seems like we can remove that complexity and just let the activity update the adapter when necessary.